File: Array/lect17_Cout_SubArray_Sum_Equal_To_K/BruteForce.cpp

#include <bits/stdc++.h>
using namespace std;

class BruteForce
{

public:
    int countSubArray(vector<int> a, int k)
    {
        int count = 0;
        int n = a.size();

        for (int i = 0; i < n; i++)
        {
            for (int j = i; j < n; j++)
            {
                int sum = 0;
                for (int K = i; K <= j; K++)
                    sum = sum + a[K];

                if (sum == k)
                    count++;
            }
        }

        return count;
    }
};

main()
{

    vector<int> arr = {1, 2, 3, -3, 1, 1, 1, 4, 2, -3};
    BruteForce a;
    int ans = a.countSubArray(arr, 3);
    cout << ans << endl;
}

File: Array/lect7_mejorityElement/BruteForce.java
public class BruteForce {
    public static void main(String[] args) {
        int[] array = { 2, 2, 3, 3, 1, 3, 3};

        for (int i = 0; i < array.length; i++) {
            int count = 0;
            for (int j = 0; j < array.length; j++) {
                if (array[j] == array[i]) {
                    count++;
                }
            }
            if (count > (array.length) / 2) {
                System.out.println(array[i]);
                break;
            } 
            
        }

    }
}
